src/Nix/Exec.hs

```diff
@@ -178,13 +179,11 @@ instance MonadNix e m => MonadEval (NValue m) m where
         span  <- currentPos
         pure $ nvConstantP (Provenance scope (NConstant_ span c)) c
 
-    evalString = assembleString >=> \case
-        Just (s, c) -> do
-            scope <- currentScopes
-            span  <- currentPos
-            pure $ nvStrP (Provenance scope
-                           (NStr_ span (DoubleQuoted [Plain s]))) s c
-        Nothing -> nverr $ ErrorCall "Failed to assemble string"
+    evalString ns = do
+        scope <- currentScopes
+        span  <- currentPos
+        pure $ nvStrP (Provenance scope
+                           (NStr_ span (DoubleQuoted [Plain (stringIntentionallyDropContext ns)]))) ns
```

Comment:
Hmm it's hard to tell here what `evalString` is doing... @jwiegley What's this used for? We definitely shouldn't drop context in a generic function...

Reply:
`evalString` is recording provenance while building a string. Sounds like context needs to be preserved here.